cdap-ui/app/cdap/components/PreviewData/DataView/Table.tsx

```diff
@@ -15,61 +15,78 @@
  */
 
 import React from 'react';
-import Table from '@material-ui/core/Table';
-import TableBody from '@material-ui/core/TableBody';
-import TableHead from '@material-ui/core/TableHead';
-import TableRow from '@material-ui/core/TableRow';
 import Paper from '@material-ui/core/Paper';
-import TableCell from '@material-ui/core/TableCell';
-import withStyles, { WithStyles } from '@material-ui/core/styles/withStyles';
+import Grid from '@material-ui/core/Grid';
+import withStyles, { WithStyles, StyleRules } from '@material-ui/core/styles/withStyles';
 import isEmpty from 'lodash/isEmpty';
 import { PREVIEW_STATUS } from 'services/PreviewStatus';
 import Heading, { HeadingTypes } from 'components/Heading';
 import ThemeWrapper from 'components/ThemeWrapper';
+import VirtualScroll from 'components/VirtualScroll';
 import T from 'i18n-react';
+import classnames from 'classnames';
 
 const I18N_PREFIX = 'features.PreviewData.DataView.Table';
 
-export const CustomTableCell = withStyles((theme) => ({
-  head: {
-    backgroundColor: theme.palette.grey['300'],
-    color: theme.palette.common.white,
-    padding: 10,
-    fontSize: 14,
-    '&:first-of-type': {
-      borderRight: `1px solid ${theme.palette.grey['500']}`,
-    },
-  },
-  body: {
-    padding: 10,
-    fontSize: 14,
-    '&:first-of-type': {
-      borderRight: `1px solid ${theme.palette.grey['500']}`,
-    },
-  },
-}))(TableCell);
-
 export const messageTextStyle = {
   fontSize: '1.3rem !important',
   margin: '10px 0',
 };
-export const styles = (theme) => ({
+export const styles = (theme): StyleRules => ({
   root: {
-    width: '100%',
     display: 'inline-block',
     height: 'auto',
     marginTop: theme.spacing(1),
   },
+  tableContainer: {
+    width: 'fit-content',
+    minWidth: '100%',
+  },
+  recordContainer: {
+    width: '100%',
+  },
   table: {
     width: '100%',
   },
   row: {
     height: 40,
-    '&:nth-of-type(odd)': {
+    '&.oddRow': {
       backgroundColor: theme.palette.grey['600'],
     },
   },
   messageText: messageTextStyle,
+  headerRow: {
+    backgroundColor: theme.palette.grey['300'],
+    fontWeight: 500,
+  },
+  headerCell: {
+    color: theme.palette.common.white,
+    fontSize: 14,
+  },
+  cell: {
+    textAlign: 'center',
```

Comment:
nit: Can we have the text align to left? I believe we it left aligned in other places in UI.

Reply:
Done!